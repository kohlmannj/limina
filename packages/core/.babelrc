{
  "moduleId": "Limina",
  "presets": [
    "@limina/babel-preset-core"
  ]
}
