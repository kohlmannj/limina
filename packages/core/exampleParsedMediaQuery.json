{
  "after": "",
  "before": "",
  "nodes": [],
  "sourceIndex": 0,
  "type": "media-query-list",
  "value": "@media (min-width: 480px)"
}
