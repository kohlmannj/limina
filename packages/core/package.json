{
  "name": "@limina/core",
  "version": "0.4.0",
  "description": "A CSS-in-JS library for interpolating and constraining CSS property values across screen widths",
  "main": "dist/index.js",
  "module": "lib/index.js",
  "types": "lib/index.d.ts",
  "repository": "https://github.com/kohlmannj/limina/packages/core",
  "author": "Joseph Kohlmann <joseph.kohlmann@nytimes.com>",
  "license": "Apache-2.0",
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "docs",
    "lib",
    "LICENSE",
    "README.md"
  ],
  "dependencies": {
    "csstype": "^2.6.4",
    "lodash.difference": "^4.5.0",
    "lodash.merge": "^4.6.1",
    "ml-regression-simple-linear": "^2.0.1",
    "mq-polyfill": "^1.1.8",
    "postcss-media-query-parser": "^0.2.3",
    "units-css": "^0.4.0"
  },
  "gitHead": "5bf75ff47081672d5fd06ca6c101f5e6fd3c99ad"
}
