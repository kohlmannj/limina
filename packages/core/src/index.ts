export { default } from './limina';
export * from './index.d';
