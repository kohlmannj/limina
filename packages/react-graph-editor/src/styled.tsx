import styled, { CreateStyled } from 'react-emotion';
import { IThemeProps } from './types';

export default styled as CreateStyled<IThemeProps>;
