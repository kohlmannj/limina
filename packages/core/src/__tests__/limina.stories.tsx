/* eslint-disable import/no-extraneous-dependencies */
import React, { Fragment } from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { storiesOf } from '@storybook/react';
import { createBreakpoints, limina } from '..';

storiesOf('core', module)
  .add('with div that decreases in width on wider screens', () => {
    const [mobile, tablet, desktop] = createBreakpoints([540, 854, 1280]);

    const LiminaDiv = styled('div')(
      limina({
        width: [mobile(540), tablet(320), desktop(1160)],
        fontSize: [mobile(32), tablet(48), desktop(64)],
      }),
      // Just to get syntax highlighting with the vscode-styled-components extension
      css`
        margin: 0 auto;
        background-color: rebeccapurple;
        color: goldenrod;
        font-family: sans-serif;
      `
    );

    return (
      <LiminaDiv>
        <h1>Hello World!</h1>
        <h2>Resize the preview pane to see some magic happen {'\u2728'}</h2>
      </LiminaDiv>
    );
  })
  .add('with multiple containers', () => {
    const [sm, md, ml, lg, xl] = createBreakpoints([320, 768, 1024, 1280, 1440]);

    const Container = styled('div')`
      display: flex;
      height: 200px;
      align-items: center;
      justify-content: center;
      margin: 0 auto 20px;
      background-color: orange;
      font-family: sans-serif;
    `;

    const LargeContainer = styled(Container)(
      limina({
        width: [sm(300), md(720), ml(945), lg(945), xl(1200)],
      })
    );

    const MediumContainer = styled(Container)(
      limina({
        width: [sm(300), md(630), ml(600), lg(600), xl(630)],
      })
    );

    const SmallContainer = styled(Container)(
      limina({
        width: [sm(300), md(300), ml(300), lg(300), xl(300)],
      })
    );

    return (
      <Fragment>
        <LargeContainer>
          <span>Large Container</span>
        </LargeContainer>

        <MediumContainer>
          <span>Medium Container</span>
        </MediumContainer>

        <SmallContainer>
          <span>Small Container</span>
        </SmallContainer>
      </Fragment>
    );
  });
