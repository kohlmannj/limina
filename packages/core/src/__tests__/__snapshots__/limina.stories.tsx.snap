// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core with div that decreases in width on wider screens 1`] = `
.emotion-0 {
  margin: 0 auto;
  background-color: rebeccapurple;
  color: goldenrod;
  font-family: sans-serif;
}

@media (max-width:540px) {
  .emotion-0 {
    width: 100vw;
    font-size: 5.9259259259259265vw;
  }
}

@media (min-width:541px) and (max-width:853px) {
  .emotion-0 {
    width: calc(-70.06369426751591vw + 918.343949044586px);
    font-size: calc(5.095541401273886vw + 4.484076433121018px);
  }
}

@media (min-width:854px) {
  .emotion-0 {
    width: 37.470725995316165vw;
    font-size: 5.620608899297424vw;
  }
}

@media (min-width:855px) and (max-width:1279px) {
  .emotion-0 {
    width: calc(197.1830985915493vw + -1363.9436619718308px);
    font-size: calc(3.755868544600939vw + 15.924882629107984px);
  }
}

@media (min-width:1280px) {
  .emotion-0 {
    width: 90.625vw;
    font-size: 5vw;
  }
}

<div
  class="emotion-0"
>
  <h1>
    Hello World!
  </h1>
  <h2>
    Resize the preview pane to see some magic happen âœ¨
  </h2>
</div>
`;

exports[`core with multiple containers 1`] = `
Array [
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 200px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto 20px;
  background-color: orange;
  font-family: sans-serif;
}

@media (max-width:320px) {
  .emotion-0 {
    width: 93.75vw;
  }
}

@media (min-width:321px) and (max-width:767px) {
  .emotion-0 {
    width: calc(93.75vw + 0px);
  }
}

@media (min-width:768px) {
  .emotion-0 {
    width: 93.75vw;
  }
}

@media (min-width:769px) and (max-width:1023px) {
  .emotion-0 {
    width: calc(87.890625vw + 45px);
  }
}

@media (min-width:1024px) {
  .emotion-0 {
    width: 92.28515625vw;
  }
}

@media (min-width:1025px) and (max-width:1279px) {
  .emotion-0 {
    width: calc(0vw + 945px);
  }
}

@media (min-width:1280px) {
  .emotion-0 {
    width: 73.828125vw;
  }
}

@media (min-width:1281px) and (max-width:1439px) {
  .emotion-0 {
    width: calc(159.375vw + -1095px);
  }
}

@media (min-width:1440px) {
  .emotion-0 {
    width: 83.33333333333334vw;
  }
}

<div
    class="emotion-0"
  >
    <span>
      Large Container
    </span>
  </div>,
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 200px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto 20px;
  background-color: orange;
  font-family: sans-serif;
}

@media (max-width:320px) {
  .emotion-0 {
    width: 93.75vw;
  }
}

@media (min-width:321px) and (max-width:767px) {
  .emotion-0 {
    width: calc(73.66071428571429vw + 64.28571428571428px);
  }
}

@media (min-width:768px) {
  .emotion-0 {
    width: 82.03125vw;
  }
}

@media (min-width:769px) and (max-width:1023px) {
  .emotion-0 {
    width: calc(-11.71875vw + 720px);
  }
}

@media (min-width:1024px) {
  .emotion-0 {
    width: 58.59375vw;
  }
}

@media (min-width:1025px) and (max-width:1279px) {
  .emotion-0 {
    width: calc(0vw + 600px);
  }
}

@media (min-width:1280px) {
  .emotion-0 {
    width: 46.875vw;
  }
}

@media (min-width:1281px) and (max-width:1439px) {
  .emotion-0 {
    width: calc(18.75vw + 360px);
  }
}

@media (min-width:1440px) {
  .emotion-0 {
    width: 43.75vw;
  }
}

<div
    class="emotion-0"
  >
    <span>
      Medium Container
    </span>
  </div>,
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 200px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto 20px;
  background-color: orange;
  font-family: sans-serif;
}

@media (max-width:320px) {
  .emotion-0 {
    width: 93.75vw;
  }
}

@media (min-width:321px) and (max-width:767px) {
  .emotion-0 {
    width: calc(0vw + 300px);
  }
}

@media (min-width:768px) {
  .emotion-0 {
    width: 39.0625vw;
  }
}

@media (min-width:769px) and (max-width:1023px) {
  .emotion-0 {
    width: calc(0vw + 300px);
  }
}

@media (min-width:1024px) {
  .emotion-0 {
    width: 29.296875vw;
  }
}

@media (min-width:1025px) and (max-width:1279px) {
  .emotion-0 {
    width: calc(0vw + 300px);
  }
}

@media (min-width:1280px) {
  .emotion-0 {
    width: 23.4375vw;
  }
}

@media (min-width:1281px) and (max-width:1439px) {
  .emotion-0 {
    width: calc(0vw + 300px);
  }
}

@media (min-width:1440px) {
  .emotion-0 {
    width: 20.833333333333336vw;
  }
}

<div
    class="emotion-0"
  >
    <span>
      Small Container
    </span>
  </div>,
]
`;
